### ISSUE TITLE
FRCNNResnet101 无法收敛或是loss_bbox 始终为0

### ISSUE THREAD

**Issue by SuperPowerLF2** [huaze555/windows-caffe-faster-rcnn#13]
用的models\FRCNN\res101 train_val.proto进行训练
以下是一些训练片段 感觉loss不正常 
训练完成后使用test_frcnn.exe, 直接用训练集当做测试集
结果都是 0 
详细调试后发现score < 0.8的地方 score 值非常低0.00x最高我看到的也就0.05

> I0828 11:53:16.375608 19192 solver.cpp:218] Iteration 19920 (2.24295 iter/s, 8.91684s/20 iters), loss = 1.99799
I0828 11:53:16.375608 19192 solver.cpp:237] Train net output #0: bbox_accuracy = 0.75
I0828 11:53:16.376610 19192 solver.cpp:237] Train net output #1: loss_bbox = 0.430361 (* 1 = 0.430361 loss)
I0828 11:53:16.376610 19192 solver.cpp:237] Train net output #2: loss_cls = 1.27145 (* 1 = 1.27145 loss)
I0828 11:53:16.376610 19192 solver.cpp:237] Train net output #3: rpn_cls_loss = 0.898545 (* 1 = 0.898545 loss)
I0828 11:53:16.376610 19192 solver.cpp:237] Train net output #4: rpn_loss_bbox = 0.391415 (* 1 = 0.391415 loss)
I0828 11:53:16.376610 19192 sgd_solver.cpp:105] Iteration 19920, lr = 0.001
I0828 11:53:25.249524 19192 solver.cpp:218] Iteration 19940 (2.25401 iter/s, 8.87307s/20 iters), loss = 1.99201
I0828 11:53:25.249524 19192 solver.cpp:237] Train net output #0: bbox_accuracy = 0.75
I0828 11:53:25.250700 19192 solver.cpp:237] Train net output #1: loss_bbox = 0.644284 (* 1 = 0.644284 loss)
I0828 11:53:25.250700 19192 solver.cpp:237] Train net output #2: loss_cls = 0.905329 (* 1 = 0.905329 loss)
I0828 11:53:25.250700 19192 solver.cpp:237] Train net output #3: rpn_cls_loss = 0.0850422 (* 1 = 0.0850422 loss)
I0828 11:53:25.250700 19192 solver.cpp:237] Train net output #4: rpn_loss_bbox = 0.0457095 (* 1 = 0.0457095 loss)
I0828 11:53:25.251526 19192 sgd_solver.cpp:105] Iteration 19940, lr = 0.001
I0828 11:53:34.160838 19192 solver.cpp:218] Iteration 19960 (2.24479 iter/s, 8.90953s/20 iters), loss = 2.00955
I0828 11:53:34.160838 19192 solver.cpp:237] Train net output #0: bbox_accuracy = 0.75
I0828 11:53:34.161839 19192 solver.cpp:237] Train net output #1: loss_bbox = 0.721698 (* 1 = 0.721698 loss)
I0828 11:53:34.161839 19192 solver.cpp:237] Train net output #2: loss_cls = 1.35276 (* 1 = 1.35276 loss)
I0828 11:53:34.161839 19192 solver.cpp:237] Train net output #3: rpn_cls_loss = 0.107642 (* 1 = 0.107642 loss)
I0828 11:53:34.161839 19192 solver.cpp:237] Train net output #4: rpn_loss_bbox = 0.0488492 (* 1 = 0.0488492 loss)
I0828 11:53:34.161839 19192 sgd_solver.cpp:105] Iteration 19960, lr = 0.001
I0828 11:53:43.032603 19192 solver.cpp:218] Iteration 19980 (2.25453 iter/s, 8.87101s/20 iters), loss = 1.98453
I0828 11:53:43.032603 19192 solver.cpp:237] Train net output #0: bbox_accuracy = 0.96875
I0828 11:53:43.033603 19192 solver.cpp:237] Train net output #1: loss_bbox = 0.000514256 (* 1 = 0.000514256 loss)
I0828 11:53:43.033603 19192 solver.cpp:237] Train net output #2: loss_cls = 0.411482 (* 1 = 0.411482 loss)
I0828 11:53:43.033603 19192 solver.cpp:237] Train net output #3: rpn_cls_loss = 0.192493 (* 1 = 0.192493 loss)
I0828 11:53:43.033603 19192 solver.cpp:237] Train net output #4: rpn_loss_bbox = 0.0110174 (* 1 = 0.0110174 loss)
I0828 11:53:43.033603 19192 sgd_solver.cpp:105] Iteration 19980, lr = 0.001

查看入口data 处你写死了batch_size大小1 于是我 改动了 voc_config中的batch_size 改为1 后
loss_cls 能正常缩小. 但是loss_bbox的值从第一步就是0.

> I0829 14:22:08.621503   712 solver.cpp:218] Iteration 0 (0 iter/s, 2.2033s/20 iters), loss = 3.89008
I0829 14:22:08.621503   712 solver.cpp:237]     Train net output #0: bbox_accuracy = 0
I0829 14:22:08.622505   712 solver.cpp:237]     Train net output #1: loss_bbox = 0 (* 1 = 0 loss)
I0829 14:22:08.622505   712 solver.cpp:237]     Train net output #2: loss_cls = 3.04452 (* 1 = 3.04452 loss)
I0829 14:22:08.622505   712 solver.cpp:237]     Train net output #3: rpn_cls_loss = 0.693147 (* 1 = 0.693147 loss)
I0829 14:22:08.622505   712 solver.cpp:237]     Train net output #4: rpn_loss_bbox = 0.203805 (* 1 = 0.203805 loss)
I0829 14:22:08.622505   712 sgd_solver.cpp:105] Iteration 0, lr = 0.001
I0829 14:22:23.028703   712 solver.cpp:218] Iteration 20 (1.38832 iter/s, 14.4059s/20 iters), loss = 3.72456
I0829 14:22:23.028703   712 solver.cpp:237]     Train net output #0: bbox_accuracy = 1
I0829 14:22:23.029705   712 solver.cpp:237]     Train net output #1: loss_bbox = 0 (* 1 = 0 loss)
I0829 14:22:23.029705   712 solver.cpp:237]     Train net output #2: loss_cls = 2.81629 (* 1 = 2.81629 loss)
I0829 14:22:23.029705   712 solver.cpp:237]     Train net output #3: rpn_cls_loss = 0.68107 (* 1 = 0.68107 loss)
I0829 14:22:23.029705   712 solver.cpp:237]     Train net output #4: rpn_loss_bbox = 0.203163 (* 1 = 0.203163 loss)
I0829 14:22:23.029705   712 sgd_solver.cpp:105] Iteration 20, lr = 0.001

请问如何修改才能恢复正常! 谢谢!

**Comment by huaze555** [comment]
@SuperPowerLF2 你好，这个项目我已经不再维护了。建议使用yolov3

**Comment by SuperPowerLF2** [comment]
@huaze555 谢谢回复
